cupyx/scipy/signal/_waveforms.py

```diff
@@ -0,0 +1,690 @@
+                raise ValueError(
+                    "Reference level for time cutoff must " "be < 0 dB")
+            tref = pow(10.0, tpr / 20.0)
+            return np.sqrt(-np.log(tref) / a)
+        else:
+            raise ValueError("If `t` is a string, it must be 'cutoff'")
+
+    t = cupy.asarray(t)
+
+    if not retquad and not retenv:
+        return _gausspulse_kernel_F_F(t, a, fc)
+    if not retquad and retenv:
+        return _gausspulse_kernel_F_T(t, a, fc)
+    if retquad and not retenv:
+        return _gausspulse_kernel_T_F(t, a, fc)
+    if retquad and retenv:
+        return _gausspulse_kernel_T_T(t, a, fc)
+
+
+_chirp_phase_lin_kernel_real = cupy.ElementwiseKernel(
+    "T t, T f0, T t1, T f1, T phi",
+    "T phase",
+    """
+    const T beta { (f1 - f0) / t1 };
+    const T temp { 2 * M_PI * (f0 * t + 0.5 * beta * t * t) };
+    // Convert  phi to radians.
+    phase = cos(temp + phi);
+    """,
+    "_chirp_phase_lin_kernel",
+    options=("-std=c++11",),
+)
+
+_chirp_phase_lin_kernel_cplx = cupy.ElementwiseKernel(
+    "T t, T f0, T t1, T f1, T phi",
+    "Y phase",
+    """
+    const T beta { (f1 - f0) / t1 };
+    const T temp { 2 * M_PI * (f0 * t + 0.5 * beta * t * t) };
+    // Convert  phi to radians.
+    phase = Y(cos(temp + phi), cos(temp + phi + M_PI/2) * -1);
+    """,
+    "_chirp_phase_lin_kernel",
+    options=("-std=c++11",),
+)
+
+_chirp_phase_quad_kernel = cupy.ElementwiseKernel(
+    "T t, T f0, T t1, T f1, T phi, bool vertex_zero",
+    "T phase",
+    """
+    T temp {};
+    const T beta { (f1 - f0) / (t1 * t1) };
+    if ( vertex_zero ) {
+        temp = 2 * M_PI * (f0 * t + beta * (t * t * t) / 3);
+    } else {
+        temp = 2 * M_PI *
+            ( f1 * t + beta *
+            ( ( (t1 - t) * (t1 - t) * (t1 - t) ) - (t1 * t1 * t1)) / 3);
+    }
+    // Convert  phi to radians.
+    phase = cos(temp + phi);
+    """,
+    "_chirp_phase_quad_kernel",
+    options=("-std=c++11",),
+)
+
+_chirp_phase_log_kernel = cupy.ElementwiseKernel(
+    "T t, T f0, T t1, T f1, T phi",
+    "T phase",
+    """
+    T temp {};
+    if ( f0 == f1 ) {
+        temp = 2 * M_PI * f0 * t;
+    } else {
+        T beta { t1 / log(f1 / f0) };
+        temp = 2 * M_PI * beta * f0 * ( pow(f1 / f0, t / t1) - 1.0 );
+    }
+    // Convert  phi to radians.
+    phase = cos(temp + phi);
+    """,
+    "_chirp_phase_log_kernel",
+    options=("-std=c++11",),
+)
+
+_chirp_phase_hyp_kernel = cupy.ElementwiseKernel(
+    "T t, T f0, T t1, T f1, T phi",
+    "T phase",
+    """
+    T temp {};
+    if ( f0 == f1 ) {
+        temp = 2 * M_PI * f0 * t;
+    } else {
+        T sing { -f1 * t1 / (f0 - f1) };
+        temp = 2 * M_PI * ( -sing * f0 ) * log( abs( 1 - t / sing ) );
+    }
+    // Convert  phi to radians.
+    phase = cos(temp + phi);
+    """,
+    "_chirp_phase_hyp_kernel",
+    options=("-std=c++11",),
+)
+
+
+def chirp(t, f0, t1, f1, method="linear", phi=0, vertex_zero=True,
+          type="real"):
```

Comment:
Seems SciPy (stable 1.11.2 and development) does not have this argument. Where does it come from?

Reply:
Ah, okay. I didn't check cuSignal implementation. Thanks!